notebooks/photutils/03_aperture_photometry/03_aperture_photometry.ipynb

```diff
@@ -0,0 +1,970 @@
+    "               rotation=270, labelpad=30)\n",
+    "ax1.set_xlabel('X (pixels)')\n",
+    "ax1.set_ylabel('Y (pixels)')\n",
+    "ax1.set_title('Elliptical Apertures')\n",
+    "\n",
+    "# Crop to show an inset of the data\n",
+    "ax1.set_xlim(2000, 3000)\n",
+    "ax1.set_ylim(2000, 1000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Clearly, these custom-made elliptical apertures fit our XDF galaxies much better than the one-size-fits-all circular apertures from before."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Sky Coordinates & Apertures\n",
+    "\n",
+    "At the moment, the positions of our apertures are in pixels, relative to our data array. However, if you need aperture positions in terms of celestial coordinates, `photutils` also includes aperture objects that can be integrated with Astropy's `SkyCoords`.\n",
+    "\n",
+    "Fortunately this is extremely easy when we use the [World Coordinate System (WCS)](http://docs.astropy.org/en/stable/wcs/) to decipher a WCS object from the header of the FITS file containing our image, and then the `to_sky()` method to transform our `EllipticalAperture` objects into `SkyEllipticalAperture` objects. \n",
+    "\n",
+    "And, better yet, our `CCDData` object has held onto its WCS information ever since we created it!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs = xdf_image.wcs\n",
+    "sky_elliptical_apertures = [ap.to_sky(wcs) for ap in elliptical_apertures]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we wanted to generate `SkyEllipticalAperture` objects from the get-go, we could have used that WCS object in the following way:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from photutils import SkyEllipticalAperture\n",
+    "from astropy.coordinates import SkyCoord\n",
+    "\n",
+    "r = 3.  # pixels; approximate isophotal extent of semimajor axis\n",
+    "\n",
+    "# Create the apertures\n",
+    "sky_elliptical_apertures = []\n",
+    "for obj in catalog:\n",
+    "    # Convert the centroids into RA/Dec using WCS\n",
+    "    ra, dec = wcs.all_pix2world(obj.xcentroid.value, obj.ycentroid.value, 0)\n",
+    "    # Convert the positions to an Astropy SkyCoord object, with units!\n",
+    "    sky_position = SkyCoord(ra, dec, unit=u.deg)\n",
+    "    \n",
+    "    # Define the elliptical parameters, now with units\n",
+    "    a = obj.semimajor_axis_sigma.value * r * u.pix\n",
+    "    b = obj.semiminor_axis_sigma.value * r * u.pix\n",
+    "    theta = obj.orientation.value  * u.rad\n",
+    "    \n",
+    "    # Convert the theta from radians from X axis to the radians from North \n",
+    "    x_to_north_angle = (90. + header['ORIENTAT']) * u.deg\n",
```

Comment:
I get `NameError: name 'header' is not defined`, but cannot see anywhere in the notebook that `header` is previously called. Should there be a cell somewhere which has the same `with file as f: header = blah.header` bit as I recall from a previous notebook?